notebooks/3.jump-analysis/nbconverted/3.jump_analysis.py

```diff
@@ -423,28 +433,90 @@
+col_arrangement = [
+    "Metadata_Plate",
+    "Plate_Map_Name",
+    "Metadata_Well",
+    "Cell_type",
+    "Metadata_pert_type",
+    "Time",
+    "Metadata_gene",
+    "Metadata_pert_iname",
+    "Metadata_target_sequence",
+    "pred_injury",
+    "probability",
+]
+
+# creaeting a rename dict
+rename_assay_type = {
+    "JUMP-Target-1_orf_platemap": "orf",
+    "JUMP-Target-1_compound_platemap": "compound",
+    "JUMP-Target-1_crispr_platemap": "crispr",
+}
+
 # split meta and feature columns
 shared_jump_meta, shared_jump_feats = split_meta_and_features(shared_jump_df)
 
 # selecting the columns
-jump_meta = shared_jump_df[["Metadata_Plate", "Metadata_Well", "Metadata_pert_iname"]]
+predicted_df = shared_jump_df[
+    [
+        "Metadata_Plate",
+        "Metadata_Well",
+        "Metadata_gene",
+        "Metadata_pert_iname",
+        "Metadata_target_sequence",
+        "Metadata_pert_type",
+    ]
+]
 
 # converting injury codes to injury names
-jump_meta["pred_injury"] = [
+predicted_df["pred_injury"] = [
     injury_decoder[str(injury_code)] for injury_code in y_pred.tolist()
 ]
 
 # obtaining the probability score of the predicted injury
-jump_meta["probability"] = y_proba.max(axis=1).tolist()
+predicted_df["probability"] = y_proba.max(axis=1).tolist()
 
+# add experimental metadata
+sel_jump_exp_meta = jump_exp_meta[["Assay_Plate_Barcode", "Cell_type", "Time"]]
+predicted_df = predicted_df.merge(
+    sel_jump_exp_meta, left_on="Metadata_Plate", right_on="Assay_Plate_Barcode"
+).drop(columns=["Assay_Plate_Barcode"])
 
-# In[20]:
+# Merge barcode information by using the Plate ID to indicate the type of assay conducted
+predicted_df = pd.merge(
+    predicted_df, barcode_df, left_on="Metadata_Plate", right_on="Assay_Plate_Barcode"
+)
+predicted_df = predicted_df.drop("Assay_Plate_Barcode", axis=1)
+predicted_df = predicted_df[col_arrangement].rename(
+    columns={"Plate_Map_Name": "Assay_type"}
+)
 
 
-# save supplemental table
-jump_meta.to_csv(
-    jump_analysis_dir / "stable2_predicted_jump_injury_table.csv.gz",
-    compression="gzip",
-    index=False,
+# updating column containign assay information
+predicted_df["Assay_type"] = predicted_df["Assay_type"].apply(
+    lambda assay_code: rename_assay_type[assay_code]
 )
-jump_meta
+
+# if 'Metadata_pert_type' is NaN this means that no sample was added
+# rational behind this is because the "Metadata_broad_sample" is empty indicating no sample was added
+predicted_df["Metadata_pert_type"] = predicted_df["Metadata_pert_type"].fillna("EMPTY")
+
+# update 'trt' value to "treatment" to improve readability
+predicted_df["Metadata_pert_type"] = predicted_df["Metadata_pert_type"].apply(
+    lambda name: "treatment" if name == "trt" else name
+)
+
+# dropping duplicates
+predicted_df = predicted_df.drop_duplicates()
+
+# display
+print(predicted_df.shape)
+predicted_df.head()
+
+
+# In[20]:
+
+
+predicted_df.to_csv(jump_analysis_dir / "JUMP_predicted_injuries.csv", index=False)
```

Comment:
We'd benefit in the future from calling this `JUMP_CP_Pilot_predicted_injuries.csv`, just so we're clear we are analyzing the pilot data only

Reply:
(if you make this change, please make sure to delete the old file from this repo to avoid confusion)

Reply:
Done will update this too in the supplemental